### DISCUSSION TITLE
Compilation failed on Ubuntu 20.04 with python 3.8

### DISCUSSION THREAD

**Discussion by MingxuanChe** [se-hwan/cusadi#5] [General]
Hi,

First I would like to thank you for open-sourcing your work. I am very interested in it and would like to try it out.

Unfortunately, I found a compilation error when following the provided installation steps. The error is as follows:
`Generating double code`
`Generating CUDA code for CasADi function:  inertial_quantities`
`CUDA codegen complete for CasADi function:  inertial_quantities`
`inertial_quantities`
`Compiling CUDA code...`
`sh: 1: cd: can't cd to build`
`Compilation failed.`

I use Ubuntu 20.04, Python 3.8, casadi 3.6.5, cuda 12.2 and I have exported the PATH and LD_LIBRARY_PATH.

I feel like it might be a CMake-related issue since the `build` folder is not created. I have CMake 3.30.1

Could you give some hints on how could I fix this? Thank you in advance!

Best,
Mingxuan

**Comment by se-hwan** [comment]
Hi Mingxuan, 

From the error, it seems like you didn't make a `build` directory in the root folder for `cusadi`?
You need to do this manually with `mkdir build` before doing any codegen. 

Hope that helps!
